doc/normalization.rst

```diff
@@ -0,0 +1,114 @@
+Gardener Component Descriptor Normalization
+===========================================
+
+Every resource can contain an optional digest attribut that specifies the digest of the referenced resource.
+The digest is an object that *MUST* contain the `algorithm` and the hash as `value`. Additionally, it *MUST* contain the normalisationAlgorithm.
+
+.. code-block::
+  meta:
+    schemaVersion: 'v2'
+
+  component:
+    name: 'example.com/my-component'
+    version: 'v1.1.1'
+
+    resources:
+    - name: my-artifact
+      version: v0.0.0
+      digest:
+        algorithm: sha256
+        normalisationAlgorithm: manifest-digest-v1
+        value: 'abcdef'
+
+
+Normalization
+--------------
+
+The normalized component descriptor has the following properties:
+- Attributes *MUST* be transformed into list of single attributes (key-value pairs) and ordered alphabetically
+- The component descriptor *MUST* be JSON encoded
+- Component References are included with their identity fields (name, version and extra identity) and their content digest.
+  The content digest contains algorithm, normalisationAlgorithm and value.
+- Resources contain their identity fields (name, version and extra identity) as well as their content digest.
+- Sources are included only their identity fields
+  - Sources without a dedicated artifact can be added to the normalized version by using the `extraIdentity` field.
+
+The following listing shows the list of single attributes in the normalised version.
+.. code-block::
+  - meta:
+    - schemaVersion: 'v2'
+
+  - component:
+      - name: 'example.com/my-component'
+      - version: 'v1.1.1'
+      - digest:
+        - algorithm: sha256
+        - normalisationAlgorithm: sorted-alphabetically-v1
+        - value: 'abcdef'
```

Comment:
digest?

Reply:
i would prefer value, otherwise the access would be digest.digest, this way it is digest.value. wdyt?